cranelift/codegen/src/isa/x64/inst/encoding/mod.rs

```diff
@@ -1 +1,57 @@
+use crate::{isa::x64, machinst::MachBuffer};
+use std::vec::Vec;
+
+pub mod evex;
 pub mod rex;
+pub mod vex;
+
+pub trait CodeSink {
```

Comment:
Yes, [here](https://github.com/bytecodealliance/wasmtime/blob/aa259ff92a4cb04762b88f9485f817995ecbc680/cranelift/codegen/src/binemit/mod.rs#L127-L171) but as you can see it would force this module to implement things it has no idea about and I wasn't too sure the old backend trait would be around forever. Also, I am trying to keep this module as dependency-free as possible so that in the future it could be used elsewhere--if I want to use this in the future to just encode instructions I won't want or need the additional `MachBuffer` methods and fields.

Reply:
Maybe we could name the trait something else -- `ByteSink` or something?

It might even make sense to do the factoring in the other direction -- split out `ByteSink` from `CodeSink` and make the former a constraint on the latter (`trait CodeSink : ByteSink { ... }` with just the additional methods), but that's probably out-of-scope for this PR...